src/rezplugins/build_system/python.py

```diff
@@ -0,0 +1,325 @@
+                            varname = "__PARSE_ARG_%s" % key.upper()
+
+                            # do some value conversions
+                            if isinstance(value, bool):
+                                value = 1 if value else 0
+                            elif isinstance(value, (list, tuple)):
+                                value = map(str, value)
+                                value = map(quote, value)
+                                value = ' '.join(value)
+
+                            executor.env[varname] = value
+
+                if env:
+                    for key, value in env.items():
+                        if isinstance(value, (list, tuple)):
+                            for path_ in value:
+                                executor.env[key].append(path_)
+                        else:
+                            executor.env[key] = value
+
+            return _callback
+
+        build_arg_parser = argparse.ArgumentParser()
+        build_arg_parser.add_argument('--develop')
+
+        build_args = self.build_args or []
+        install_mode = install
+        develop_mode = 'develop' in build_args
+        pip_mode = 'pip' in  build_args
+
+        source_path = self.working_dir
+        py_install_root = os.path.join(source_path, 'build', '_py_install')
+        py_develop_root = os.path.join(source_path, 'build', '_py_develop')
+        dist_root = os.path.join(source_path, 'dist')
+        setup_py = os.path.join(source_path, 'setup.py')
+        prefix = 'rez'
+
+        def _run_context_shell(command, cwd, env=None):
+            _callback = _make_callack(env)
+            retcode, _, _ = context.execute_shell(command=command,
+                                                  block=True,
+                                                  cwd=cwd,
+                                                  parent_environ=None,
+                                                  actions_callback=_callback)
+            return retcode
+
+        def _copy_tree(src, dest):
+            # Utility function to copy tree.
+            if os.path.exists(dest):
+                shutil.rmtree(dest)
+            if os.path.exists(src):
+                shutil.copytree(src, dest)
+
+        def _setup_py_build():
+            cmds = ['python', setup_py, 'install', '--root', py_install_root, '--prefix', prefix, '-f']
+            # subprocess.call(cmds, cwd=source_path)
+            return _run_context_shell(cmds, cwd=source_path)
+
+        def _pip_build():
+            #TODO: pip 10 will install wheel builds by default, making this intermediate wheel build step unnecessary
+            # Generate wheel
+            print('Building wheel...')
+            pip = 'pip'
+            # These args speed up pip builds
+            pip_args = ['--disable-pip-version-check', '--no-deps', '--no-index']
+            cmds = [pip, 'wheel', '.', '-w', dist_root] + pip_args
+            # subprocess.call(cmds, cwd=source_path)
```

Comment:
Is this for reference or is it just what was used before `_run_context_shell(cmds, cwd=source_path)`?

Reply:
It's an undeleted reference comment.  This code was originally a utility library for the bez build system using rezbuild.py.  I basically broke apart that utility script and inserted it into a rez build plugin skeleton, making some changes to get it to work correctly.  This is one of the parts that need to be changed to work with the build plugin and the original code from the utility script just got left in by accident.

Reply:
Hopefully I'll get some time this month to revisit this branch and clean some stuff like this up.  If you seen any other areas that can be improved, please let me know, thanks.

Reply:
Cool. Yeah, even though Git has marked it all as new, I figured that wasn't completely true. I get that you're just making it play nice with setup.py so please take my comments with a grain of salt. If you get time to clean up some of the commented code then great, otherwise no worries at all.